epl_trends/process_json_data.js

```diff
@@ -0,0 +1,179 @@
+	var allJsonArrays = Array.prototype.slice.call(arguments); // convert arguments to a normal array
+	allJsonArrays.splice(0,1); // chop off the error part of arguments so you are left w. just the q.defer results
+	allJsonArrays.forEach(function(d, i) {
+		var current_season = seasons[i];
+	})
+
+	var merged_data = d3.merge(allJsonArrays);
+
+	merged_data.sort(function(a, b) { 
+		return d3.ascending(a.startyear, b.startyear); 
+	});
+	
+	// this is where you put code to use merged_data!!!!!!!!!!!!!!!!!!!!!!
+	var yvar = document.getElementById("yaxis").value.toLowerCase();
+	plotarea = addDataToPlot(plotarea, merged_data, yvar);
+	plotarea = update(plotarea);
+	plotarea = exit(plotarea);
+
+	d3.select("form").on("submit", function(d) {
+		
+        var new_yvar = document.getElementById("yaxis").value.toLowerCase();
+        if(new_yvar == "goal differential") {
+        	new_yvar = "goals-dff";
+        }
+
+        plotarea = addDataToPlot(plotarea, merged_data, new_yvar);
+        
+        //d3.selectAll(".datapt").exit().remove();
+		//d3.selectAll(".teamline").exit().remove();
+		console.log(d3.selectAll(".datapt"));
+
+		//by default html form submissions does a full page refresh, this turns that off
+        d3.event.preventDefault(); 
+
+    });
+
+	return plotarea;
+}
+
+function addDataToPlot(plot, data, yvar) {
+	
+	// add to scaling functions based on data
+	var ymax = d3.max(data, function(d) { return d[yvar]; })
+		uniqueyears = d3.map(data, function(d){ return d.startyear; }).keys()
+		uniqueteams = d3.map(data, function(d){ return d.team; }).keys();
+	xScale.domain(uniqueyears); 
+	yScale.domain([ymax, 0]);
+	colorScale.domain(uniqueteams);
+
+	// add axes
+	axesDetails.append("g")
+          .attr("id", "xAxis")
+          .attr("transform", "translate(" + 0 + "," + plotheight + ")")
+          .classed("axis", true)
+          .call(xAxis);
+    axesDetails.append("g")
+          .attr("id", "yAxis")
+          .classed("axis", true)
+          .call(yAxis);
+
+	// setup line function
+	var buildLine = d3.line().curve(d3.curveCardinal)
+		.x(function(d) { return xScale(d.startyear); })
+		.y(function(d) { return yScale(d[yvar]); });
+
+	// organize data by team
+	var databyteam = d3.nest()
+    	.key(function(d) { return d.team; })
+    	.entries(data);
+    
+	// add points
+	var pts_g = plot.append("g").attr("class", "datapt");
+	pts_g.selectAll(".datapt")
+		.data(data).enter()
+		.append("circle")
+		.attr("class", "datapt")
+		.attr("r", 3)
+		.attr("cx", function(d) { return xScale(d.startyear); })
+		.attr("cy", function(d) { return yScale(d[yvar]); })
+		.style("fill", function(d) { return colorScale(d.team); });
+		
+	// add lines
+	var lines_g = plot.append("g").attr("class", "teamline")	
+	databyteam.forEach(function(d, i) {
+		lines_g.append("path")
+		.attr("class", "teamline")
+		.attr("d", buildLine(d.values))
+		.style("stroke", colorScale(d.key))
+		.style("fill", "none");
+	})
```

Comment:
i'm probably missing something important, and am not testing this, but maybe something like:
```
var lines = lines_g.
  .data(databyteam)
  .append("path")
  .attr("class", "teamline")
  .attr("d", function(d) { return buildLine(d.values) })
  .etc.
```
does that work? seems like `.data()` and selections ought to be useful for iterating over elements of databyteam